bigframes/ml/decomposition.py

```diff
@@ -197,3 +198,177 @@ def score(
 
         # TODO(b/291973741): X param is ignored. Update BQML supports input in ML.EVALUATE.
         return self._bqml_model.evaluate()
+
+
+@log_adapter.class_logger
+class MF(
+    base.UnsupervisedTrainablePredictor,
+    bigframes_vendored.sklearn.decomposition._mf.MF,
+):
+    __doc__ = bigframes_vendored.sklearn.decomposition._mf.MF.__doc__
+
+    def __init__(
+        self,
+        n_components: Optional[Union[int, float]] = None,
+        *,
+        num_factors: int,
+        user_col: str,
+        item_col: str,
+        l2_reg: float,
```

Comment:
Per: https://cloud.google.com/bigquery/docs/reference/standard-sql/bigqueryml-syntax-create-matrix-factorization#l2_reg

* The default value is 1.0.
* This could also be a `HPARAM_RANGE(1.5, 5.0)` or `HPARAM_CANDIDATES([0, 1.0, 3.0, 5.0])`. Let's add a TODO to support hyperparameter tuning. (Maybe in a subsequent PR).

```suggestion
        # TODO: Add support for hyperparameter tuning.
        l2_reg: float = 1.0,
```